### ISSUE TITLE
Use cube texture as a storage type

### ISSUE THREAD

**Issue by JonahPlusPlus** [gpuweb/gpuweb#3331]
I'm writing a procedural sky compute shader that creates a sky texture. I am using a 6x1 2D texture, but now I want to move it to a cube texture to get better filtering (in low resolutions, the mesh is visible, although the frag shader is unlit; my UVs are shrinked by 1 pixel to prevent bleed but that seems to be limited).
However, I can't textureStore to a cube texture. Instead, I get the error `cube dimension is not expected for texture storage`. What am I supposed to do? I thought this was possible in GLSL; are there plans to make this available? Is there a workaround available?

**Comment by JonahPlusPlus** [comment]
The following is the issue I get with a 2D 6:1 texture (8x8px per side):
![image](https://user-images.githubusercontent.com/33059163/184713984-ae9545d1-9f5b-400e-8c40-a1fd9358f540.png)
The problem disappears as you increase the resolution (32x32px per side):
![image](https://user-images.githubusercontent.com/33059163/184714188-f7f008b1-4419-4c50-a8a1-89860cae6c95.png)
Sure, I can just use a higher resolution, but I would prefer it if my users had the flexibility to use whatever without worry.

**Comment by kainino0x** [comment]
A texture can only be 1d/2d/3d, while a texture view can be cube or not. So, your texture has GPUTextureDimension ["2d"](https://gpuweb.github.io/gpuweb/#dom-gputexturedimension-2d) and 6 (or more) array layers. You can create multiple views from it: one with GPUTextureViewDimension ["2d-array"](https://gpuweb.github.io/gpuweb/#dom-gputextureviewdimension-2d-array) for writing as storage, and one with ["cube"](https://gpuweb.github.io/gpuweb/#dom-gputextureviewdimension-cube) for sampling during rendering.

Does this resolve your issue?

**Comment by JonahPlusPlus** [comment]
Kinda? I worked on doing that before I saw your message. (I'm using bevy btw)
The problem now is that while the corners are gone, I can't resize the texture (I also know now that the texture wasn't resizing before; those images aren't really 8x8 or 32x32, they just make the corner more obvious at "lower" resolutions).
I'm not sure if this is a bevy issue or wgpu issue. The image says it's resized, the texture gets recreated, and yet, the texture stays the same resolution.

My code can be viewed at: [JonahPlusPlus/bevy_atmosphere](https://github.com/JonahPlusPlus/bevy_atmosphere)

The following is an 8x8 texture that has been resized to 512x512 (the corners are obvious but that not a problem to me, it's just a result of the extremely low resolution). It's supposed to be a daylight cycle, but when I change resolutions, it breaks and stays static:
![image](https://user-images.githubusercontent.com/33059163/185016952-e31e9210-98bb-46e4-bf35-4adb95d309aa.png)
